internal/nhooyr.io/websocket/conn_options.go

```diff
@@ -0,0 +1,30 @@
+package websocket
+
+import (
+	"errors"
+	"time"
+)
+
+var (
+	ErrMisconfiguredWS = errors.New("misconfigured WS")
+)
+
+// Option is a functional option type for WS.
+type ConnOption interface {
+	apply(*Conn)
+}
+
+type ConnOptionFunc func(*Conn)
+
+func (f ConnOptionFunc) apply(c *Conn) {
+	f(c)
+}
+
+// InactivityTimeout sets inactivity timeout for the WS connection.
+// Nonpositive InactivityTimeout will default to a 1 minute timeout.
+func InactivityTimeout(d time.Duration) ConnOption {
```

Comment:
We used the options patterns to set `inactivityTimeout`, in order to avoid more locks due to the `inactivityTimeout` race condition created in `c.timeoutLoop()`:

```
c.wg.Add(1)
go func() {
	defer c.wg.Done()
	c.timeoutLoop()
}()
```